ts/Series/GeoHeatmap/GeoHeatmapPoint.ts

```diff
@@ -0,0 +1,78 @@
+/* *
+ *
+ *  (c) 2010-2023 Highsoft AS
+ *  Authors:
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import MapPoint from '../Map/MapPoint.js';
+import GeoHeatmapPointOptions from '../GeoHeatmap/GeoHeatmapPointOptions.js';
+import GeoHeatmapSeries from '../GeoHeatmap/GeoHeatmapSeries.js';
+import { PointShortOptions } from '../../Core/Series/PointOptions.js';
+import U from '../../Core/Utilities.js';
+const {
+    isNumber
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+class GeoHeatmapPoint extends MapPoint {
+
+    public options: GeoHeatmapPointOptions = void 0 as any;
+    public series: GeoHeatmapSeries = void 0 as any;
+
+    public applyOptions(
+        options: (GeoHeatmapPointOptions|PointShortOptions),
+        x?: number
+    ): GeoHeatmapPoint {
+        const point = super.applyOptions.call(this, options, x),
+            lat = point.options.lat,
+            lon = point.options.lon;
+
+        if (isNumber(lon) && isNumber(lat)) {
+            const colsize = this.series.options.colsize || 1,
+                rowsize = this.series.options.rowsize || 1,
+                x1 = lon - colsize / 2,
+                y1 = lat - rowsize / 2;
+
+            point.geometry = point.options.geometry = {
+                type: 'Polygon',
+                // A rectangle centered in lon/lat
+                coordinates: [
+                    [
+                        [x1, y1],
+                        [x1 + colsize, y1],
+                        [x1 + colsize, y1 + rowsize],
+                        [x1, y1 + rowsize]
```

Comment:
I think (but this must be tested) that we need to add `[x1, y1]` at the end of the polygon too. Otherwise the renderer will draw a straight line back to the origin instead of a geodesic curve.